packages/types/src/codec/createType.ts

```diff
@@ -94,6 +95,9 @@ export function getTypeDef (_type: Text | string): TypeDef {
   } else if (startingWith(type, 'Compact<', '>')) {
     value.info = TypeDefInfo.Compact;
     value.sub = getTypeDef(subType);
+  } else if (startingWith(type, 'Option<', '>')) {
```

Comment:
I believe we are missing a "from the info, create the type" around line 117 in this file.